### PR TITLE
Stop masked data in MDWorkspaces being displayed in slice viewer

### LINKED ISSUE #14523: Stop masked regions of MDWorkspaces being drawn in the SliceViewer
Masks currently do not affect how MDWorkspaces are displayed in the Slice Viewer.

Display in the same way as NaNs or 0s?

This issue is related to a general review of masks: #12689.

### PR DESCRIPTION
**Pull request by matthew-d-jones**
Closes #14523 

Masked data in MDWorkspaces now is now displayed with a value of 0 in the Slice Viewer. This is consistent with MatrixWorkspaces.

**To Test:**
Use the following script to make a masked and unmasked workspace of each MDWorkspace type (MDHistoWorkspace and MDEventWorkspace).
Compare the appearance of masked and unmasked of each type in the Slice Viewer. The central unit cube of the masked workspaces should be masked and data cursor should show a value of 0 ("Intensity=" in the top right of the viewer window).

``` python

ws_mdhw = CreateMDWorkspace(Dimensions='3', Extents='-2,2,-2,2,-2,2', Names='h,k,l',
                       Units='rlu,rlu,rlu', SplitInto='12')
FakeMDEventData(ws_mdhw, UniformParams='1000', RandomizeSignal='1')
ws_mdhw = BinMD(ws_mdhw,AlignedDim0='h,-2,2,12',AlignedDim1='k,-2,2,12',AlignedDim2='l,-2,2,12')
ws_masked_mdhw = CloneMDWorkspace(ws_mdhw)
MaskMD(Workspace=ws_masked_mdhw,Dimensions="h,k,l",Extents="-1,1,-1,1,-1,1")

ws_mdew = CreateMDWorkspace(Dimensions='3', Extents='-2,2,-2,2,-2,2', Names='h,k,l',
                       Units='rlu,rlu,rlu', SplitInto='12')
FakeMDEventData(ws_mdew, UniformParams='1000', RandomizeSignal='1')
ws_masked_mdew = CloneMDWorkspace(ws_mdew)
MaskMD(Workspace=ws_masked_mdew,Dimensions="h,k,l",Extents="-1,1,-1,1,-1,1")

```

### CODE REVIEW

File: Framework/DataObjects/test/MDEventWorkspaceTest.h [hunk 1]
```diff
@@ -295,6 +295,39 @@ class MDEventWorkspaceTest : public CxxTest::TestSuite {
...
+
+    std::vector<coord_t> min;
+    std::vector<coord_t> max;
+
+    min.push_back(0);
+    min.push_back(0);
+    min.push_back(0);
+    max.push_back(1.5);
+    max.push_back(1.5);
+    max.push_back(1.5);
```
Comment (picatess) [inline Framework/DataObjects/test/MDEventWorkspaceTest.h]: You could use initializer list now, but that is not really important here. See [here](http://en.cppreference.com/w/cpp/language/list_initialization)

File: Framework/DataObjects/test/MDEventWorkspaceTest.h [hunk 2]
```diff
@@ -295,6 +295,39 @@ class MDEventWorkspaceTest : public CxxTest::TestSuite {
...
+
+    // Create an function to mask some of the workspace.
+    MDImplicitFunction *function = new MDBoxImplicitFunction(min, max);
```
Comment (picatess) [inline Framework/DataObjects/test/MDEventWorkspaceTest.h]: does ew take charge of the implicit function, ie does it `delete` the implicit function? If so fine, else you need to free the data.

Comment (matthew-d-jones) [inline Framework/DataObjects/test/MDEventWorkspaceTest.h]: It does `delete` the function.

File: Framework/API/src/MatrixWorkspace.cpp [hunk 3]
```diff
@@ -1601,6 +1601,20 @@ signal_t MatrixWorkspace::getSignalAtCoord(
...
+* @param coords :: bare array, size 2, of coordinates. X, Y
+* @param normalization :: how to normalize the signal
+* @return normalized signal.
+*/
+signal_t MatrixWorkspace::getSignalWithMaskAtCoord(
+    const coord_t *coords,
+    const Mantid::API::MDNormalization &normalization) const {
+  return getSignalAtCoord(coords, normalization);
+}
+
```
Comment (picatess) [inline Framework/API/src/MatrixWorkspace.cpp]: Why do you not check for the mask here?

**Comment by picatess** [comment]
Followed test instructions. Worked as indicated. The masked bins are set to 0. Happy to merge

**Pull request merged**